### PR TITLE
Add test case 'commit_with_backing'

### PR DESCRIPTION
**Pull request by lolyu**
test qemu-img commit with option '-b', which specifies the layer to
commit to.

Signed-off-by: lolyu <lolyu@redhat.com>

### CODE REVIEW

**Comment by lolyu** [comment]
```
 (01/50) repeat1.Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.commit_with_backing: PASS (301.89 s)
```

**Comment by lolyu** [comment]
```
PASS 1-Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.commit_with_backing
```

**Comment by lolyu** [comment]
@hellohellenmao please take a review

**Comment by lolyu** [comment]
@hellohellenmao @hereischen please take a review

File: qemu/tests/qemu_disk_img.py [hunk 1]
```diff
@@ -46,6 +46,26 @@ def create_snapshot(self, t_params=None):
         self.trash.append(snapshot)
         return params
 
+    @error_context.context_aware
+    def commit(self, drop=False, cache_mode=None, base=None):
+        """
+        Commit snapshot to base file
+        """
+        error_context.context("commit snapshot")
+        cmds = [self.image_cmd]
```
Comment (hereischen) [inline qemu/tests/qemu_disk_img.py]: `cmds = [self.image_cmd, "commit"]`?

Comment (lolyu) [reply qemu/tests/qemu_disk_img.py]: Nice one.

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by hellohellenmao** [comment]
Pass rate 48/50 for "ERROR: Server said: 'RSS_CREATE_FILE: error receiving or writing file contents'" error.

**Comment by lolyu** [comment]
@hellohellenmao please elaborate the error reproduction env here.

**Comment by hellohellenmao** [comment]
# python3 ConfigTest.py --testcase=commit_with_backing --guestname=Win2019 --platform=x86_64 --nicmodel=virtio_net --driveformat=virtio_scsi --imageformat=qcow2 --clone=no --nrepeat=50

Error info:
13:21:18 ERROR| Reproduced traceback from: /usr/local/lib/python3.6/site-packages/avocado_plugins_vt-70.0-py3.6.egg/avocado_vt/test.py:449
13:21:18 ERROR| Traceback (most recent call last):
13:21:18 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/commit_with_backing.py", line 55, in run
13:21:18 ERROR|     hashes[guest_file] = image.imgobj.save_file(guest_file)
13:21:18 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-70.0-py3.6.egg/virttest/error_context.py", line 135, in new_fn
13:21:18 ERROR|     return fn(*args, **kwargs)
13:21:18 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/qemu_disk_img.py", line 127, in save_file
13:21:18 ERROR|     self.__create_file(dst)
13:21:18 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-70.0-py3.6.egg/virttest/error_context.py", line 135, in new_fn
13:21:18 ERROR|     return fn(*args, **kwargs)
13:21:18 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/qemu_disk_img.py", line 102, in __create_file
13:21:18 ERROR|     self.vm.copy_files_to(src, dst)
13:21:18 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-70.0-py3.6.egg/virttest/error_context.py", line 135, in new_fn
13:21:18 ERROR|     return fn(*args, **kwargs)
13:21:18 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-70.0-py3.6.egg/virttest/virt_vm.py", line 1243, in copy_files_to
13:21:18 ERROR|     filesize=filesize)
13:21:18 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-70.0-py3.6.egg/virttest/remote.py", line 882, in transfer
13:21:18 ERROR|     ret = func(*args, **kwargs)
13:21:18 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-70.0-py3.6.egg/virttest/remote.py", line 930, in copy_files_to
13:21:18 ERROR|     c.upload(local_path, remote_path, timeout)
13:21:18 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-70.0-py3.6.egg/virttest/rss_client.py", line 357, in upload
13:21:18 ERROR|     raise FileTransferServerError(errmsg)
13:21:18 ERROR| virttest.rss_client.FileTransferServerError: Server said: 'RSS_CREATE_FILE: error receiving or writing file contents'

**Comment by lolyu** [comment]
```
 (1/1) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.commit_with_backing: PASS (1218.25 s)
```

**Comment by lolyu** [comment]
thanks all.

**Pull request merged**